### PR TITLE
Add extent to local mesh

### PR DESCRIPTION
**Pull request by streeve**
Add `extent` of the locally owned mesh (with or without ghosted cells) for convenience. This matches the global mesh `extent` and will be useful for the grid communication features.

### CODE REVIEW

File: cajita/src/Cajita_LocalMesh.hpp [hunk 1]
```diff
@@ -120,6 +120,19 @@ class LocalMesh<Device, UniformMesh<Scalar>>
         return _ghost_high_corner[dim];
     }
 
+    // Get the extent of a given dimension.
+    KOKKOS_INLINE_FUNCTION
+    Scalar extent( Own, const int dim ) const
+    {
+        return _own_high_corner[dim] - _own_low_corner[dim];
```
Comment (dalg24) [inline cajita/src/Cajita_LocalMesh.hpp:127]: I would provide a templated version that calls `highCorner` and `lowCorner`

Comment (sslattery) [reply cajita/src/Cajita_LocalMesh.hpp:127]: You want a single function for both instantiations?

Comment (dalg24) [reply cajita/src/Cajita_LocalMesh.hpp:127]:
```C++
template <typename OwnOrGhost>
KOKKOS_FUNCTION Scalar extent( OwnOrGhost w, const int dim ) const
{
       return highCorner(w, dim) - lowCorner(w, dim);
}
```

Comment (sslattery) [reply cajita/src/Cajita_LocalMesh.hpp:127]: oh yeah good call

**Comment by streeve** [comment]
Ok, I added your suggestion

File: cajita/src/Cajita_LocalMesh.hpp [hunk 2]
```diff
@@ -120,6 +120,13 @@ class LocalMesh<Device, UniformMesh<Scalar>>
         return _ghost_high_corner[dim];
     }
 
+    // Get the extent of a given dimension.
+    template <typename OwnOrGhost>
```
Comment (sslattery) [inline cajita/src/Cajita_LocalMesh.hpp:124]: Change `OwnOrGhost` to `Decomposition` to be consistent with other classes.

**Pull request merged**